### PR TITLE
Updated packages and dotnet standard

### PR DESCRIPTION
**Pull request by DennisAMenace**
We where targeting netstandard2.0 and should be netstandard2.1 for dotnet 3
https://docs.microsoft.com/en-us/dotnet/standard/net-standard

### CODE REVIEW

File: src/NBitcoin.Tests/NBitcoin.Tests.csproj [hunk 1]
```diff
@@ -19,7 +19,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.9.0" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.5.0-preview-20200203-01" />
```
Comment (dangershony) [inline src/NBitcoin.Tests/NBitcoin.Tests.csproj]: Is it a good idea to go preview?

File: src/Stratis.Bitcoin.Features.BlockStore/Stratis.Bitcoin.Features.BlockStore.csproj [hunk 2]
```diff
@@ -21,12 +21,12 @@
     <PackageLicenseExpression>MIT</PackageLicenseExpression>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="DBreeze" Version="1.89.0" />
-    <PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="3.1.0" />
-    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="3.1.0" />
-    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="3.1.0" />
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="3.1.0" />
-    <PackageReference Include="Tracer.Fody" Version="3.2.0">
+    <PackageReference Include="DBreeze" Version="1.95.0" />
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.BlockStore/Stratis.Bitcoin.Features.BlockStore.csproj]: Personally I am not sure we should just upgrade all nugets to latest ver.
Did you try to run networks?

Comment (DennisAMenace) [reply src/Stratis.Bitcoin.Features.BlockStore/Stratis.Bitcoin.Features.BlockStore.csproj]: I did, and have not run into any issues. I've run x42 for about a month with this new version and it's stable.

File: src/NBitcoin.Tests/NBitcoin.Tests.csproj (hunk 1 above)
Comment (DennisAMenace) [reply src/NBitcoin.Tests/NBitcoin.Tests.csproj]: Good Catch, Thank you.

File: src/Stratis.Bitcoin.Features.BlockStore/Stratis.Bitcoin.Features.BlockStore.csproj (hunk 2 above)
Comment (dangershony) [reply src/Stratis.Bitcoin.Features.BlockStore/Stratis.Bitcoin.Features.BlockStore.csproj]: ah ok thats good news then, I suggest if we can wait til my coindb refactor goes in, then I will try out the changes in this PR together with the optimization PR to sync BTC and STRAT.

File: src/Stratis.Bitcoin.Api.Tests/Stratis.Bitcoin.Api.Tests.csproj [hunk 3]
```diff
@@ -9,12 +9,12 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.9.0" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.5.0-preview-20200203-01" />
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Api.Tests/Stratis.Bitcoin.Api.Tests.csproj]: There are a few more places with preview lets fix them all.

Comment (DennisAMenace) [reply src/Stratis.Bitcoin.Api.Tests/Stratis.Bitcoin.Api.Tests.csproj]: Fixed them all here, sorry about that.

File: src/Stratis.Bitcoin.Features.Consensus/ConsensusController.cs [hunk 4]
```diff
@@ -83,13 +73,13 @@ public IActionResult DeploymentFlags()
...
-        [Route("api/[controller]/getbestblockhash")]
+        [Route("getbestblockhash")]
         [HttpGet]
-        public IActionResult GetBestBlockHashAPI()
+        public IActionResult GetBestBlockHash()
         {
             try
             {
-                return this.Json(this.GetBestBlockHashRPC());
+                return this.Json(this.ChainState.ConsensusTip?.HashBlock);
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.Consensus/ConsensusController.cs]: seems to me this breaks the previous return type no?

Comment (DennisAMenace) [reply src/Stratis.Bitcoin.Features.Consensus/ConsensusController.cs]: This is a result of the ConsensusController split to separate API & RPC controller's as the attribute for the API controllers that are required in the new version of swagger break the RPC.

Comment (dangershony) [reply src/Stratis.Bitcoin.Features.Consensus/ConsensusController.cs]: ah gotcha ok fine

**Comment by dangershony** [comment]
I just wanted to focus a second on this comment
> We where targeting netstandard2.0 and should be netstandard2.1 for dotnet 3   

I don't think there is a requirement to support netstandard2.1 when upgrading to netcore 3
If you refer to the comment
> The following table lists the minimum platform versions that support each .NET Standard version. That means that later versions of a listed platform also support the corresponding .NET Standard version. For example, .NET Core 2.2 supports .NET Standard 2.0 and earlier.  

This means netcore 3 supports netstandard2.1 and lower

**Comment by DennisAMenace** [comment]
> I just wanted to focus a second on this comment
> 
> > We where targeting netstandard2.0 and should be netstandard2.1 for dotnet 3
> 
> I don't think there is a requirement to support netstandard2.1 when upgrading to netcore 3
> If you refer to the comment
> 
> > The following table lists the minimum platform versions that support each .NET Standard version. That means that later versions of a listed platform also support the corresponding .NET Standard version. For example, .NET Core 2.2 supports .NET Standard 2.0 and earlier.
> 
> This means netcore 3 supports netstandard2.1 and lower

I will remove this if you want.

**Comment by dangershony** [comment]
No this was more of an observation to discuss not a call to action.
The general rule is for a lib (in this case Stratis.Bitcoin) try to support as low as possible netstandard version so the lib will work with a wider set of libraries. 

Perhaps it does not matter because most projects are netcore anyway

**Pull request merged**